### PR TITLE
Deterministic test (Chainer)

### PR DESCRIPTION
**Pull request by niboshi**
This is Chainer counterpart of https://github.com/cupy/cupy/pull/82.

(See also: https://github.com/cupy/cupy/pull/222)

### CODE REVIEW

File: chainer/testing/random.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                try:
+                    impl(self, *args, **kw)
+                finally:
+                    _teardown_random()
+            return test_func
+        elif isinstance(impl, type) and issubclass(impl, unittest.TestCase):
+            # Applied to test case class
+            klass = impl
+
+            def wrap_setUp(f):
```
Comment (okuta) [inline chainer/testing/random.py]: How about using simple way?
``` python
_setUp = klass.setUp
@functools.wraps(_setUp)
def setUp(self):
    _setup_random()
    _setUp(self)
klass.setUp = setUp
```

File: chainer/testing/random.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                def func(self):
```
Comment (okuta) [inline chainer/testing/random.py]: How about using `@functools.wraps`?

File: chainer/testing/random.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                    _setup_random()
+                    f(self)
+                return func
+
+            def wrap_tearDown(f):
```
Comment (okuta) [inline chainer/testing/random.py]: ditto

**Comment by niboshi** [comment]
Thank you for suggestion! Fixed.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**